File: 14-longest-common-prefix/longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string longest = strs[0];

        for(int i = 1 ; i < strs.size() ;i++){
            int j = 0 , k = 0 ;
            // longets = "";
            string s = "";
            string match = strs[i];
            while(j < longest.size() && k < match.size()){
                if(longest[j] == match[k]){
                    s.push_back(match[k]);
                }else{
                    break;
                }
                j++;
                k++;
            }
            if(s.size() == 0)return "";

            longest = s ;
        }
        return longest ;
    }
};

File: 14-longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        Arrays.sort(strs);



        String fst = strs[0];
        String lst = strs[strs.length - 1];
        int i = 0;
        int j = 0;

        StringBuilder sb = new StringBuilder();

        while (i < fst.length() && j < lst.length()) {

            if (fst.charAt(i) == lst.charAt(j)) {
                sb.append(fst.charAt(i));
                i++;
                j++;
            }

            else {
                break;
            }

        }

        return sb.toString(); 
    }
}